hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieCatalog.java

```diff
@@ -345,7 +345,7 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable catalogTable, boo
       options.put(TableOptionProperties.PARTITION_COLUMNS, partitions);
 
       final String[] pks = conf.getString(FlinkOptions.RECORD_KEY_FIELD).split(",");
-      boolean complexHoodieKey = pks.length > 1 || resolvedTable.getPartitionKeys().size() > 1;
+      boolean complexHoodieKey = pks.length > 1 || partitions.split(",").length > 1;
```

Comment:
What's the difference?

Reply:
`ResolvedTable. getPartitionKeys()` returns a list. `newArrayList ("par1, par2")`, where the length is actually 1 and the element is "par1, par2"

Reply:
It should not be the case, otherwise code L343 ~ L345 does not work well.

Reply:
I think there's no problem. The partitions field is also linked by commas, and it's reasonable for me to check the number of partitions separated by commas. If you think it's unreasonable, could you please provide a demo so that I can reflect on it
<img width="838" alt="iShot_2024-03-18_11 41 55" src="https://github.com/apache/hudi/assets/28925346/a3bd438a-10e9-4369-a86f-f9b99bb15d90">

Reply:
@danny0405  Hi, we have found a friendly way to modify it and have made the necessary changes. Could you please help me take a look? Thank you